app/family-history/family-history.service.js

```diff
@@ -0,0 +1,268 @@
+          params.payload.contact_id +
+          ',"' +
+          params.payload.contact_date +
+          '",' +
+          params.payload.contact_type +
+          ',' +
+          params.payload.contact_status +
+          ',' +
+          params.payload.reason_not_contacted +
+          ',"' +
+          params.payload.remarks +
+          '")';
+      }
+      queryParts = {
+        sql: sql
+      };
+      return db.queryServer(queryParts, function (result) {
+        result.sql = sql;
+        resolve(result);
+      });
+    });
+  };
+
+  getContactTracingHistory = (params) => {
+    return new Promise((resolve, reject) => {
+      let queryParts = {};
+      let sql = `select t2.*, id, contact_id, contact_date, remarks,
+        case 
+          when contact_type = 1555 then @contact_type:='Phone tracing' 
+          when contact_type = 10791 then @contact_type:='Physical tracing' 
+        end as contact_type,
+        case 
+          when contact_status = 1065 then @contact_status:='Contacted' 
+          when contact_status = 1118 then @contact_status:='Not contacted'
+        end as contact_status,
+        case 
+          when reason_not_contacted = 1 then @reason_not_contacted:='Incorrect locator information' 
+          when reason_not_contacted = 2 then @reason_not_contacted:='Not found/Travelled' 
+          when reason_not_contacted = 3 then @reason_not_contacted:='Not known in the area'
+          when reason_not_contacted = 4 then @reason_not_contacted:='Relocated' 
+          when reason_not_contacted = 5 then @reason_not_contacted:='Deceased' 
+          when reason_not_contacted = 6 then @reason_not_contacted:='other'
+          when reason_not_contacted = 7 then @reason_not_contacted:='Invalid phone number'
+          when reason_not_contacted = 8 then @reason_not_contacted:='Phone off' 
+          when reason_not_contacted = 9 then @reason_not_contacted:='Wrong phone number' 
+        end as reason_not_contacted
+        from etl.contact_tracing t1 join etl.flat_family_testing t2 on (t1.contact_id = t2.obs_group_id) 
+        where obs_group_id = ${params.contact_id}`;
+      queryParts = {
+        sql: sql
+      };
+      return db.queryServer(queryParts, function (result) {
+        result.sql = sql;
+        resolve(result);
+      });
+    });
+  };
+
+  getContactObsData = (params) => {
+    return new Promise((resolve, reject) => {
+      let queryParts = {};
+      let sql =
+        "select uuid from amrs.obs where obs_id = '" + params.contact_id + "'";
+      queryParts = {
+        sql: sql
+      };
+      return db.queryServer(queryParts, function (result) {
+        result.sql = sql;
+        resolve(result);
+      });
+    });
+  };
+
+  deleteContact = (params) => {
+    let that = this;
+    return this.getContactObsData(params).then((res) => {
+      var uri = this.getRestResource(
```

Comment:
Move this to a service like obsservice in openmrs-rest folder. To make it reusable